### DISCUSSION TITLE
"max filter length limit reached" after updating to v.0.23.8

### DISCUSSION THREAD

**Discussion by andrejbroncek** [pocketbase/pocketbase#6130] [Q&A]
Hello, after updating to the latest version of Pocketbase (from previous 0.22.x version), I am seeing `max filter length limit reached` error popping up in my PB logs, apparently linked to this function used on my SvelteKit server to retrieve all service statuses for services contained on a dashboard. 

```
// Get the service status of the services used in the dashboard
const services_status = await locals.pb?.collection('service_status').getFullList({
    filter: dashboard.platforms.map((id: string) => `service_id="${id}"`).join('||'),
    expand: 'service_id, service_id.category'
});
```

There can be tens to hundreds of service id's added to the filter, so I understand the filter string can get pretty long. Is there a way to make the filter accept longer values, or am I actually approaching this in a bad way and should try to somehow rewrite the whole logic?

**Comment by chnfyi** [comment]
Hi, the max allowed individual search filter parsable length is 3500.

A very long filter can impact performance so i think you may have to split your array regarding its length and make multiple requests.

**Comment by ganigeorgiev** [comment]
The filter query param limit was introduced with [v0.23.0-rc13](https://github.com/pocketbase/pocketbase/releases/tag/v0.23.0-rc13) to minimize the risk of abusing the API and the related query execution (_we have a default query timeout limit but still it is better to not have to rely on it_). As others already mentioned, the filter length can be up to 3500 characters and in addition can have up to 200 expressions.

We can bump the limits but there will be always a limit and I'd recommend to search a different loading strategy for your use case.

If `service_status.service_id` is expected to match a single record then the easiest workaround would be to batch the requests, something like (_I haven't tested it; this could be also more efficient as the requests will be processed concurrently by the PocketBase server_):


```js
let promises = [];

let size = 100;
for (let i = 0; i < dashboard.platforms.length; i += size) {
    let ids = dashboard.platforms.slice(i, i + size);

    promises.push(locals.pb?.collection('service_status').getFullList({
        filter: ids.map((id) => locals.pb.filter("service_id={:id}", { id })).join('||'),
        expand: 'service_id, service_id.category'
    }))
}

let serviceStatuses = (await Promise.all(promises)).flat()
```
